sklearn/ensemble/gradient_boosting.py

```diff
@@ -1846,3 +1853,467 @@ def apply(self, X):
         leaves = super(GradientBoostingRegressor, self).apply(X)
         leaves = leaves.reshape(X.shape[0], self.estimators_.shape[0])
         return leaves
+
+
+class GradientBoostingClassifierCV(BaseEstimator):
+    """ Gradient Boosting Classification with Cross Validation
+
+    This class keeps adding estimators to the underlying GB estimator till
+    adding a new estimators does not affect the score by much. The goal of this
+    class is to determine the best choice of `n_estimators` for each set of
+    parameters. In most cases, this will be much faster than using
+    :class:`sklearn.model_selection.GridSearchCV` due to early stopping.
+    All parameters which are not parsed by this class are passed directly to
+    the underlying Gradient Boosting estimator. If one or more parameters is a
+    list, the Gradient Boosting model is evaluated for all elements of the
+    list.
+
+    Parameters
+    ----------
+    n_stop_rounds : int, optional, default=10
+        If the score on the test set rounded off to `score_precision` decimal
+        places does not change for `n_stop_rounds` iterations, the gradient
+        boosting is halted.
+
+    score_precision : int, optional, default=2
+        The number of decimal places to round the score off by before
+        comparing it with the scores of previous iterations.
+
+    max_iterations : int, optinal, default=10000
```

Comment:
It should not, That is the maximum iterations any of the underlying estimators should take. This number is kept large on purpose.